packages/doc/src/CommentStore.ts

```diff
@@ -0,0 +1,63 @@
+import { Config, EntityDbMetadata } from "joist-codegen";
+
+/**
+ * Not used within MarkdownCommentStore, but allows additional customisation
+ * of the Field comments by presenting the target place it will be used.
+ */
+export type FieldSourceType = "get" | "set" | "opts";
+
+export type EnumData = { name: string };
+
+/**
+ * The base API for a providing comments to be applied to the various files.
+ *
+ * Focused around Joist concepts (Entities, Fields etc) over the files themselves.
+ */
+export abstract class CommentStore {
+  constructor(protected config: Config) {}
+
+  /**
+   * Provides a comment for a Entity Field
+   */
+  abstract forField(
+    entity: EntityDbMetadata,
+    fieldName: string,
+    source: FieldSourceType,
+    generated: boolean,
```

Comment:
Ah I see, maybe call this `isCodegen` or `isCodegenFile`? I was thinking "well everything that we're outputting is generated", but we meant like the base-vs-codegen file.

I was thinking maybe the integration could inject this, instead of the comment store, but I guess the comment store is the one that knows the file name.

Wdyt of maybe always including the "generated from ..." suffix? Like even if we're writing to a codegen file, I could see someone reading the getter comments and seeing the same "ah I go to Foo.md" hint being somewhat useful. Dunno, unless you think it'd be too cluttered and we should taken the opportunity of "this is a codegen file that is obvious the comment is generated from somewhere" to skip the boilerplate.

Reply:
Hmm, yeah. I get the name issue. We could flip it to isUserland or the like.

CommentStore has been built to not assume file backing; if someone (no judgement on if this is a terrible idea) wanted to build something like `NotionCommentStore`.

I'll remove this from the CommentStore API, and just always incude it within MarkdownCommentStore. We can always add it back in if it does get cluttered.

Reply:
Another thought on this one; if we wanted it to be handled by the integration, we could have the CommentStore API return something like

`type CommentResult = { comment: string, source?: string }`

So we can easily build the string of `Generated from {source}` if given.

Reply:
Sorry for the spam; one final thought. I wondered if there's a relevant TSDoc tag to use for this. The one that perhaps best fits is `@privateRemarks` or alternatively a custom tag such as `@source`

Reply:
Agreed a custom tag like `@source` seems neat!

I like the `{ comment: string; source?: string }` API and then, yeah, let the integration decide whether it injects the source (or maybe just always inject the `@source`, even for comments in `AuthorCodegen.ts`).